standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -53,7 +53,6 @@
   private static final String JDBC_OUTPUTFORMAT_CLASS = "org.apache.hive.storage.jdbc.JdbcOutputFormat".intern();
 
   String type = null; // MYSQL, POSTGRES, ORACLE, DERBY, MSSQL, DB2 etc.
-  String driverClassName = null;
```

Comment:
This should not be removed as it is used by the providers to set the driver classname. Please add it back

Reply:
AbstractJDBCConnectorProvider Class extends AbstractDataConnectorProvider(super) Class. driverClassName is already a field in the super class. Having driverClassName=null here will overwrite the passed in driverClassName by specific provider, as AbstractJDBCConnectorProvider constructor is calling its super class constructor. This line is the root cause of "Could not find a provider for remote database" error. Removing the line will stop the overwritten problem. All the provider can set driverClassName via calling AbstractDataConnectorProvider's constructor, which is what AbstractJDBCConnectorProvider is doing for all the providers.